apps/site/assets/ts/components/Modal.tsx

```diff
@@ -48,44 +61,21 @@ const ModalContent = ({
   closeModal,
   role,
   ariaLabel,
-  focusElementId,
-  bodyPadding,
-  scrollBarPadding,
-  className = ""
-}: ModalContentProps): ReactElement<HTMLElement> => {
-  useLayoutEffect(() => {
-    // Activate trap and disable scroll on background body
-    const trap = createFocusTrap("#modal-cover", {
-      initialFocus: `#${focusElementId}`,
-      clickOutsideDeactivates: true
-    });
-    trap.activate();
-
-    const htmlElement = document.getElementsByTagName("html")[0];
-    const bodyWrapper = document.getElementById("body-wrapper");
-    if (bodyWrapper) {
-      // aria-hidden for mobile devices where we want to emulate "focus" behavior
-      bodyWrapper.setAttribute("aria-hidden", "true");
-      htmlElement.classList.add("modal-open");
-      bodyWrapper.style.paddingRight = `${scrollBarPadding}px`;
-    }
-
-    // Deactivate
-    return () => {
-      if (bodyWrapper) {
-        bodyWrapper.setAttribute("aria-hidden", "false");
-        htmlElement.classList.remove("modal-open");
-        bodyWrapper.style.paddingRight = bodyPadding;
-      }
-      trap.deactivate();
-    };
-  });
```

Comment:
If you add a dependency array of [] here (as the last arg to the hook) then I think it solves your problem without getting rid of the focus trap and adding extra logic.

You will want to do something like ` focusElementId={state.modalId === "origin" ? "origin-search" : ...}`

Reply:
incredible. Thank you so much for pointing this out!